File: go/leetcode/Array/283 - MoveZeros/moveZeros.go
package array

// https://leetcode.com/problems/move-zeroes/

func moveZeroes(nums []int) {
	nonZero := 0

	for i := range nums {
		if nums[i] != 0 {
			nums[nonZero], nums[i] = nums[i], nums[nonZero]
			nonZero++
		}
	}
}

File: python/leetcode/Array/283 - MoveZeros/moveZeros.py
# https://leetcode.com/problems/move-zeroes/

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        non_zero = 0

        for i, current in enumerate(nums):  # O(n)
            if current != 0:
                nums[i], nums[non_zero] = nums[non_zero], current  # O(1)
                non_zero += 1
